internal/version/version.go

```diff
@@ -1,7 +1,23 @@
 package version
 
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+)
+
 var (
 	BuildDate    = "unknown"
 	BuildVersion = "0.0.0"
 	Commit       = "unknown"
 )
+
+// BaseVersion returns the base version of the application.
```

Comment:
Suggestion: As a new term "the base version of the application" is introduced, it would be good to add a comment explaining it:

```suggestion
// BaseVersion returns the base version of the application.
//
// The base version is defined using the "MAJOR.MINOR" format. Patch and any metadata
// are removed.
```